configurator/main.py

```diff
@@ -0,0 +1,123 @@
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# !/usr/bin/env python
+import ast
+from distutils import dir_util
+import logging
+import shutil
+from configparserutil import ConfigParserUtil
+import os
+import constants
+from jinja2 import Environment, FileSystemLoader
+
+logging.basicConfig(level=logging.DEBUG,
+                    format='%(asctime)s %(levelname)-8s %(message)s',
+                    datefmt='%a, %d %b %Y %H:%M:%S',
+                    filename=constants.LOG_FILE_LOCATION,
+                    filemode='w')
+
+PATH = os.path.dirname(os.path.abspath(__file__))
+TEMPLATE_ENVIRONMENT = Environment(
+    autoescape=False,
+    loader=FileSystemLoader(os.path.join(PATH)),
+    trim_blocks=False)
+PACK_LOCATION = None
+
+# parse the xml file and return the content as a text
+def render_template(template_filename, context):
+    return TEMPLATE_ENVIRONMENT.get_template(template_filename).render(context)
+
+
+# This method write the output generated from jinja file to xml file
+# template_path : Path to the template file
+# output_path : Path to the output xml file
+# context : dictionary containing values to be used by jinja engine
+def create_output_xml(template_path, output_path, context):
+    # check output path exists
+    directory = os.path.dirname(output_path)
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+    with open(output_path, 'w') as xml_file:
+        content = render_template(template_path, context)
+        xml_file.write(content)
+
+
+# Read the config.ini and generate context based on settings
+# if read_env_variables is true context will be generated from environment variables
+# if read_env_variables is not true context will be read from config.ini
+def generate_context(config_file_path):
+    # Read configuration file
+    config_parser = ConfigParserUtil()
+    config_parser.read(config_file_path)
+    configurations = config_parser.as_dict()
+
+    # Reading the default values
+    context = configurations[constants.CONFIG_DEFAULTS]
+    settings = configurations[constants.CONFIG_SETTINGS]
+    global PACK_LOCATION
+    PACK_LOCATION = settings["pack_location"]
+    if settings["read_env_variables"] == "true":
+        logging.info("Reading from environment variables")
```

Comment:
Might be better to use a logger [1] than directly using "logging". Then again please have the variable name as "log".
[1] https://docs.python.org/2/howto/logging.html